### DISCUSSION TITLE
Proper way to update account balances?

### DISCUSSION THREAD

**Discussion by blathers123** [firefly-iii/firefly-iii#7826] [Firefly III]
I'm writing a script that reads email notifications of transactions from my bank and updates firefly with the parsed result. Some of the notifications are daily account balances, so every day I get a notification with the current balance of the account. I've been trying to use this to set the account balance on firefly so it's always up-to-date, but I'm running into issues. I was trying to create an opening balance transaction but I can't seem to get the api to cooperate, it doesn't like any value I set source_name to, including null. 

I get the feeling that I'm fighting the program by doing this. Is there a better way?

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/faq/financials/#how-do-i-set-the-balance-of-an-account

**Comment by blathers123** [comment]
Sorry if I'm being dumb but I'm not seeing an answer in the faq. 

Basically my question is, can/should I create an opening balance transaction every day to keep the account balance correct? Should I instead use the virtual balance as a work around?

**Reply by JC5** [reply]
From the FAQ:

**How do I set the balance of an account?**

You don't. You can only influence the balance of an account by actually making or importing transactions that influence the balance. You can't just change the balance of the account.

**But then what's the opening balance for?**

The opening balance is meant to set the balance of an account at the start of your Firefly III administration. That means that NO earlier transactions exist in your administration, and the opening balance reflects that. It summarizes all the transactions that happened before you started using Firefly III.
